java/semantickernel-api/src/main/java/com/microsoft/semantickernel/semanticfunctions/PromptTemplateConfig.java

```diff
@@ -346,26 +346,34 @@ public List<InputParameter> getParameters() {
     private final int schema;
 
     private final String type; // { get; set; } = "completion";
+
+    private final String serviceId;
     private final String description;
 
     public PromptTemplateConfig() {
-        this("", "", null);
+        this("", "", null, null);
     }
 
     public PromptTemplateConfig(CompletionConfig completionConfig) {
-        this(1, "", "", completionConfig, new InputConfig(new ArrayList<>()));
+        this(1, "", "", null, completionConfig, new InputConfig(new ArrayList<>()));
     }
 
     public PromptTemplateConfig(
             String description, String type, @Nullable CompletionConfig completionConfig) {
-        this(1, description, type, completionConfig, new InputConfig(new ArrayList<>()));
+        this(1, description, type, null, completionConfig, new InputConfig(new ArrayList<>()));
+    }
+
+    public PromptTemplateConfig(
+            String description, String type, @Nullable String serviceId, @Nullable CompletionConfig completionConfig) {
+        this(1, description, type, serviceId, completionConfig, new InputConfig(new ArrayList<>()));
     }
 
     @JsonCreator
     public PromptTemplateConfig(
             @JsonProperty("schema") int schema,
             @JsonProperty("description") String description,
             @JsonProperty("type") String type,
+            @JsonProperty("serviceId") String serviceId,
```

Comment:
The convention set by .NET is to use snake case. 

PromptTemplateConfig.cs has List<string> DefaultServices. Maybe @shawncal could weigh in on that.

Reply:
Yeah, I could not find an equivalent c# or Python implementation. I can change based on the feedback. let me know

Reply:
@shawncal @dsgrieve Looks like this is the only outstanding comment. Can we get a consensus on this so that this PR can be merged?

Reply:
@relango Apologies for the delay, few folks have been out. I'm taking a look at this now. We want to align this across all of the implementations.

Reply:
@relango Can you take a look at this PR https://github.com/microsoft/semantic-kernel/pull/2739

We want to put the `serviceId` in the completion config. The reason for this is we want to support multiple services within a semantic function and allow for the appropriate service to be selected when the prompt is executed. 

If you update your PR to follow the same pattern I can get it approved and merged.

Reply:
Thank you @markwallace-microsoft , I looked at the PR [#2739 ](https://github.com/microsoft/semantic-kernel/pull/2739) , I already have `serviceId` added to `PromptTemplateConfig` in this PR. 

The only change I see is JsonProperty name should be changed to `service_id`. I will update PR with that change. Let me know if I missed anything.

Reply:
@markwallace-microsoft I updated PR to change JsonProperty to `service_id` in `PromptTemplateConfig`. Let me know if any other changes are needed.

Reply:
@markwallace-microsoft @johnoliver  I have rebased my PR with the latest commit. Can we get it reviewed and merged? Let me know if any additional changes are needed.